traffic_ops/v3-client/session.go

```diff
@@ -353,6 +353,48 @@ func (to *Session) request(method, path string, body []byte, header http.Header)
 	return to.ErrUnlessOKOrNotModified(r, remoteAddr, err, path)
 }
 
+// requestWithoutClosingBody makes a request to Traffic Ops, trying to refresh
+// the cookie if it is expired - but only once. It determines a cookie is
+// expired not by examining the cookie itself, but by recognizing a
+// 401 Unauthorized response status code from the server. It returns, in order,
+// a pointer to an HTTP response - which, if there is no error, contains a
+// non-nil Body the caller is expected to close -, the remote address of the
+// Traffic Ops server used to service the request, and any error that occurred.
+// Note that any response code greater than or equal to 400 will result in an
+// error - and that means that it cannot be assumed that the response object is
+// nil if there is an error! This is to provide greater visibility to
+// HTTP-level errors while still giving back as much information as possible.
+func (to *Session) requestWithoutClosingBody(method, path string, body []byte, header http.Header) (*http.Response, net.Addr, error) {
```

Comment:
I don't really want us to add yet another underlying method to the client that is only used by a small fraction of top-level methods. It makes the client inconsistent and confusing. Instead, is there a way to incorporate these changes into the single shared method I've created [here](https://github.com/apache/trafficcontrol/pull/5334/files#diff-cbf10b941a6c6105c02e8e50de8be60c46c104c53f2f531f1105dffbacd49737R339)? I'm open for suggestions if you'd like to review that approach. I'd just like the top-level method bodies to be concise and use the same underlying method for consistency.

Reply:
That'll work fine. It's a much better approach, I just didn't want to go to the trouble of refactoring the whole client in this already large PR. When I finally re-base/do this, I'll use the new method.